pkg/skaffold/k8sjob/logger/log.go

```diff
@@ -164,13 +164,26 @@ func (l *Logger) streamLogsFromKubernetesJob(ctx context.Context, id, namespace
 				return false, nil
 			}
 
-			for event := range w.ResultChan() {
-				pod, ok := event.Object.(*corev1.Pod)
-				if ok {
-					podName = pod.Name
-					// TODO(aaron-prindle) add support for jobs w/ multiple pods in the future
-					break
+			done := make(chan bool)
+			go func() {
+				for event := range w.ResultChan() {
+					pod, ok := event.Object.(*corev1.Pod)
+					if ok {
+						podName = pod.Name
+						done <- true
+						break
+					}
 				}
+			}()
+
+			select {
+			case <-ctx.Done():
```

Comment:
I'm a little confuse with this `ctx.Done()`, is the intention here to cancel the context after the `for event := range w.ResultChan()` (line 169)? Or is to listen a cancellation  from other place (e.g, Ctrl+C)? If is the first case, we are missing to call a `cancel` function after the for loop.

Reply:
The intention is to listen for cancellation (the 2nd case)